content/post/2020-03-03-packageRank/index.Rmd

```diff
@@ -0,0 +1,855 @@
+---
+slug: packageRank-intro
+title: "Counting and Visualizing CRAN Downloads with packageRank (with Caveats!)"
+authors:
+  - Peter Li
+date: 2020-03-01
+tags:
+- package development
+output:
+  html_document:
+    keep_md: true
+---
+
+```{r setup, include = FALSE}
+knitr::opts_chunk$set(fig.path = "", comment = "")
+
+knitr::knit_hooks$set(
+  plot = function(x, options) {
+    hugoopts <- options$hugoopts
+    paste0(
+      "{{<figure src=",
+      '"', x, '" ',
+      if (!is.null(hugoopts)) {
```

Comment:
here we could add 

```
if (!is.null(hugoopts$title)) {
  hugoopts$alt <- huggopts$title
}
```

if in the whole post you want to use the same words for both alternative text and title (or caption if you prefer)?

Reply:
Hmm. Switching the code is not a problem. This is what I'm seeing in the Hugo HTML preview: a) with "caption =",  I see the caption (e.g, Figure ...) but it uses the same font as the text; b) with "title =", I see the caption but it uses a different font; c) with "alt =", I don't see the caption at all. 

I just want to be clear that we need to use "alt =" for the final version, right?

I can use "title =" for proofreading.

Reply:
Is there a way to create a caption for things like the regression table? I tried the below with both "title =" and "alt = " but it doesn't appear to work:

`{r ols_summary, hugoopts = list(title = "Table 3 OLS Regression Table: Bias = Popularity + Versions, October 2019"), echo = FALSE}`

Also, how do you hyperlink to different sections in the post (e.g., Computing rank percentiles)?

Reply:
I don't know about the table, probably easier to just add a paragraph after the chunk.

To link to a section I use `[link text](#header-id)`, probably `#Computing-rank-percentiles` or  `#computing-rank-percentiles` which you can find out by looking at the preview page source (or by trying both)

Alt is for having alternative text fot people using a screenreader. You might want either alt AND title, or alt AND caption, or alt AND title for one part of the caption, caption for the other part. If alt and title/caption have the same content you can use my suggestion from earlier (or tell me to implement it next week)

Reply:
Writing alternative text is challenging.